pydicom/tests/test_waveform.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2008-2020 pydicom authors. See LICENSE file for details.
+import pytest
+
+import pydicom
+from pydicom.data import get_testdata_file
+from pydicom.filereader import dcmread
+from pydicom.tests._handler_common import ALL_TRANSFER_SYNTAXES
+from pydicom.uid import (
+    ImplicitVRLittleEndian,
+    ExplicitVRLittleEndian,
+    DeflatedExplicitVRLittleEndian,
+    ExplicitVRBigEndian,
+)
+
+try:
+    import numpy as np
+    HAVE_NP = True
+except ImportError:
+    HAVE_NP = False
+
+try:
+    from pydicom.waveform_data_handlers import numpy_handler as NP_HANDLER
+    from pydicom.waveform_data_handlers.numpy_handler import generate_multiplex
+except ImportError:
+    NP_HANDLER = None
+
+
+ECG = get_testdata_file('waveform_ecg.dcm')
+
+
+@pytest.mark.skipif(HAVE_NP, reason="Numpy available")
+def test_waveform_generator_raises():
+    """Test overlay_array raises exception for all syntaxes."""
+    ds = dcmread(ECG)
+    msg = r"The waveform data handler requires numpy"
+    with pytest.raises(RuntimeError, match=msg):
+        next(ds.waveform_generator)
+
+
+@pytest.mark.skipif(not HAVE_NP, reason="Numpy not available")
+class TestDataset:
+    """Tests for dataset.waveform_generator"""
+    def test_simple(self):
+        """Simple functionality test."""
+        ds = dcmread(ECG)
+        gen = ds.waveform_generator
+        arr = next(gen)
+        arr = next(gen)
+        with pytest.raises(StopIteration):
+            next(gen)
+
+    def test_unsupported_syntax_raises(self):
+        """Test that an unsupported syntax raises exception."""
```

Comment:
Is this even possible (apart for invalid DICOM)? As far as I understand, only the transfer syntaxes you already support are allowed (this is out of interest, the test makes sense anyway).

Reply:
Hmm, a brief look at Part 5 indicates [this](http://dicom.nema.org/medical/dicom/current/output/chtml/part05/sect_A.10.html#table_A.10-2) might be a possibility.

Reply:
Ah - thanks! It helps to look at the current version of the standard. I have a downloaded version from 2018 which I usually use for lookup...